api/scanner/media_encoding/executable_worker/executable_worker.go

```diff
@@ -93,6 +119,30 @@ func (worker *FfmpegWorker) IsInstalled() bool {
 	return worker != nil
 }
 
+func (worker *CustomRawWorker) EncodeJpeg(inputPath string, outputPath string, jpegQuality int) error {
+	tmpDir, err := ioutil.TempDir("/tmp", "photoview-convert")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	args := []string{
+		worker.path,
+		inputPath,
+		outputPath,
+		fmt.Sprintf("%d", jpegQuality),
+	}
```

Comment:
I don't think that all custom programs necessarily use this format, it would be nice to have a string where the paths and quality could be inserted into, something like this:

```
/path/to/program
 -i %input_path% -o %output_path% -q %jpeg_quality%
```